topsf/inference/uhh2.py

```diff
@@ -27,117 +28,177 @@ def uhh2(self):
+            )
 
-    #
-    # categories
-    #
-
-    # category elements to combine in fit
-    years = [
-        year % 100,
-    ]
-    channels = self.config_inst.x.fit_setup["channels"]
-    pt_bins = self.config_inst.x.fit_setup["pt_bins"]
-    wp_names = self.config_inst.x.fit_setup["wp_names"]
-    fit_vars = self.config_inst.x.fit_setup["fit_vars"][0]  # TODO: do we need to support multiple vars?
+    # Use the first config as the reference
+    ref_setup = self.config_insts[0].x.fit_setup
+    ref_channels = ref_setup["channels"]
+    ref_pt_bins = ref_setup["pt_bins"]
+    ref_wp_names = ref_setup["wp_names"]
+    ref_fit_vars = ref_setup["fit_vars"][0]  # Assuming single var for now
+    ref_shape_unc = ref_setup["shape_unc"]
+
+    # Compare the rest against the reference
+    for config_inst in self.config_insts[1:]:
+        setup = config_inst.x.fit_setup
+        if setup["channels"] != ref_channels:
+            raise ValueError(f"Config {config_inst.name} has different channels.")
+        if setup["pt_bins"] != ref_pt_bins:
+            raise ValueError(f"Config {config_inst.name} has different pt_bins.")
+        if setup["wp_names"] != ref_wp_names:
+            raise ValueError(f"Config {config_inst.name} has different wp_names.")
+        if setup["fit_vars"][0] != ref_fit_vars:
+            raise ValueError(f"Config {config_inst.name} has different fit_vars[0].")
+        if setup["shape_unc"] != ref_shape_unc:
+            raise ValueError(f"Config {config_inst.name} has different shape_unc.")
+
+    # Now safe to assign
+    channels = ref_channels
+    pt_bins = ref_pt_bins
+    wp_names = ref_wp_names
+    fit_vars = ref_fit_vars
+    shape_unc = ref_shape_unc
 
     # tuples of inference categories and
     # corresponding config category names
     categories = [
         # (columnflow_name, combine_name)
         (f"{channel}__{pt_bin}__tau32_wp_{wp_name}_{region}",
-        f"bin_{channel}__{year}__{pt_bin}__tau32_wp_{wp_name}_{region}")
-        for channel, year, pt_bin, wp_name, region in itertools.product(
-            channels,
-            years,
-            pt_bins,
-            wp_names,
-            ("pass", "fail"),
-        )
+        f"bin_{channel}__{pt_bin}__tau32_wp_{wp_name}_{region}")
+        for channel in channels
+        for pt_bin in pt_bins
+        for wp_name in wp_names
+        for region in ("pass", "fail")
     ]
 
-    # add categories to inference model
     for config_cat, inference_cat in categories:
-        self.add_category(
-            inference_cat,
-            config_category=config_cat,
-            config_variable=fit_vars,
-            mc_stats="0 1 1",
-            config_data_datasets=get_process_datasets(self.config_inst, "data"),
-            # fake data from sum of MC processes
-            data_from_processes=processes,
-            # empty bins should stay empty!!
-            empty_bin_value=0.0,
+
+        var_name = fit_vars
```

Comment:
Following up from the above suggestion:
```suggestion
        var_name = fit_vars[0]
```

Maybe also add a check that `len(fit_vars) == 1`?